### PR TITLE
Add egress sni monitoring policies example

### PR DESCRIPTION
**Pull request by vadimeisenbergibm**

### CODE REVIEW

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+---
+
+The [Configure Egress Traffic using Wildcard Hosts](/docs/examples/advanced-gateways/wildcard-egress-hosts/) example
+describes how to enable TLS egress traffic for a set of hosts in a common domain, in that case `*.wikipedia.org`. This
+example extends that example to show how to configure SNI monitoring and apply policies on TLS egress traffic.
+
+{{< boilerplate before-you-begin-egress >}}
+
+*  Configure traffic to `*.wikipedia.org` by following
+   [the steps](/docs/examples/advanced-gateways/wildcard-egress-hosts#wildcard-configuration-for-arbitrary-domains) in
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: [the instructions]

Comment (frankbu) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: in the

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+$ kubectl delete logentry egress-access -n istio-system
+$ kubectl delete stdio egress-access-logger -n istio-system
+$ kubectl delete listentry requested-server-name -n istio-system
+$ kubectl delete listchecker wikipedia-checker -n istio-system
+{{< /text >}}
+
+## Monitor the SNI and the source identity, and enforce access policies based on them
+
+Since you enabled mutual TLS between the sidecar proxies and the egress gateway, you can monitor the [service identity](/docs/concepts/what-is-istio/#citadel) of the applications that access external services, and enforce policies
+based on the identities of the source of traffic.
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: of the traffic source

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+In Istio on Kubernetes, the identities are based on
+[Service Accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/). In this
+subsection, you deploy two _sleep_ containers, `sleep-us` and `sleep-canada` under two service accounts,
+`sleep-us` and `sleep-canada`, respectively. Then you define a policy that allows applications with the `sleep-us`
+identity access the English and the Spanish versions of Wikipedia, and services with `sleep-canada` identity access the
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: s/access the/to access the/g # two places

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+English and the French versions.
+
+1.  Deploy two _sleep_ containers, namely `sleep-us` and `sleep-canada`, with `sleep-us` and `sleep-canada` service
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: /namely/d

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+    {"level":"info","time":"2019-01-10T17:33:56.779842Z","instance":"egress-access.logentry.istio-system","connectionEvent":"open","destinationApp":"","requestedServerName":"es.wikipedia.org","source":"istio-egressgateway-with-sni-proxy","sourceNamespace":"default","sourcePrincipal":"cluster.local/ns/default/sa/sleep-us","sourceWorkload":"istio-egressgateway-with-sni-proxy"}
+    {"level":"info","time":"2019-01-10T17:33:57.413908Z","instance":"egress-access.logentry.istio-system","connectionEvent":"open","destinationApp":"","requestedServerName":"fr.wikipedia.org","source":"istio-egressgateway-with-sni-proxy","sourceNamespace":"default","sourcePrincipal":"cluster.local/ns/default/sa/sleep-us","sourceWorkload":"istio-egressgateway-with-sni-proxy"}
+    {{< /text >}}
+
+    Note the `requestedServerName` attribute, and `sourcePrincipal`, it must be `cluster.local/ns/default/sa/sleep-us`.
+
+1.  Define a policy that will allow access to the Wikipedia in English and Spanish for applications with the `sleep-us`
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: s/the Wikipedia/Wikipedia/

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+    service account and to the Wikipedia in English and French for applications with the `sleep-canada` service account.
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: s/the Wikipedia/Wikipedia/

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+example extends that example to show how to configure SNI monitoring and apply policies on TLS egress traffic.
+
+{{< boilerplate before-you-begin-egress >}}
+
+*  Configure traffic to `*.wikipedia.org` by following
+   [the steps](/docs/examples/advanced-gateways/wildcard-egress-hosts#wildcard-configuration-for-arbitrary-domains) in
+   [Configure Egress Traffic using Wildcard Hosts](/docs/examples/advanced-gateways/wildcard-egress-hosts/) example,
+   **with mTLS enabled**.
+
+## SNI monitoring and access policies
```
Comment (frankbu) [inline content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: This looks the same as https://preliminary.istio.io/docs/examples/advanced-gateways/wildcard-egress-hosts/#sni-monitoring-and-access-policies

Did you intend to move it here, but forgot to remove it from there?

Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: I remove it in https://github.com/istio/istio.io/pull/3058.

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md (hunk 2 above)
Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: Addressed in 0978f7c2720ffac4e91a5196cb916fbc63110bba

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md (hunk 3 above)
Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: Addressed in d3a0846945a805ee428240dc8375bbb72ef7718e

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md (hunk 4 above)
Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: Addressed 1ee9f8b1e6ab7120eb19c9864cc8f2e0b7dbd1d6

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md (hunk 5 above)
Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: Addressed in 473dae1bb91b0e646eda6d8dafa6a7c52760bb5c

File: content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md (hunk 6 above)
Comment (vadimeisenbergibm) [reply content/docs/examples/advanced-gateways/egress_sni_monitoring_and_policies/index.md]: Addressed in 473dae1

**Pull request merged**